### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jalextowle**
# Description

This PR updates the contributing guide to include some new information about running tests locally.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -72,6 +76,19 @@ make test-wasm-browser
 make test-all
 ```
 
+### Test Issues
+Some of the tests will open a fairly large number of open files. The default allowance
+for open files on most operating systems is 64, which will cause these tests to fail. The
```
Comment (albrow) [inline CONTRIBUTING.md]: Mac OS defaults to 256 and AFAIK no one else has run into any issues before. Maybe we could rephrase this as something like:

> The default maximum number of open files for some operating systems is too low for the tests to work properly. If you see an error along the lines of "Too many open files", you may need to increase this limit.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -72,6 +76,19 @@ make test-wasm-browser
 make test-all
 ```
 
+### Test Issues
```
Comment (albrow) [inline CONTRIBUTING.md]:
```suggestion
### Potential issues
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -72,6 +76,20 @@ make test-wasm-browser
 make test-all
 ```
 
+### Potential Issues
+The default number of maximum number of open files is too low in some operating systems
```
Comment (fabioberger) [inline CONTRIBUTING.md]:
```suggestion
The default maximum number of open files is too low in some operating systems
```
